specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimtags.json

```diff
@@ -0,0 +1,503 @@
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/TagIdParameter"
+          },
+          {
+            "name": "If-Match",
+            "in": "header",
+            "required": true,
+            "description": "ETag of the Tag Entity. ETag should match the current entity state from the header response of the GET request or it should be * for unconditional update.",
+            "type": "string"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "The tag was successfully deleted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "TagCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TagContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Tag list representation."
+    },
+    "TagContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagContractProperties",
+          "description": "Tag entity contract properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./apimanagement.json#/definitions/Resource"
+        }
+      ],
+      "description": "Tag Contract details."
+    },
+    "TagContractProperties": {
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "Tag name.",
+          "maxLength": 160,
+          "minLength": 1
+        }
+      },
+      "required": [
+        "displayName"
+      ],
+      "description": "Tag contract Properties."
+    },
+    "TagCreateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagContractProperties",
+          "description": "Properties supplied to Create Tag operation."
+        }
+      },
+      "description": "Parameters supplied to the Create Tag operation."
+    },
+    "TagUpdateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagContractProperties",
```

Comment:
is displayName a required property in the UpdateContract. if yes... then we dont need 2 contracts. you can just have one contract for both Create and Update.